apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/SamhsaMatcherR4FromClaimTransformerV2Test.java

```diff
@@ -0,0 +1,751 @@
+  private static final String NON_SAMHSA_HCPCS_CODE = "11111";
+  private static final String BLACKLISTED_IC9_DIAGNOSIS_CODE = "291.0";
+  private static final String BLACKLISTED_IC10_DIAGNOSIS_CODE = "F10.10";
+  private static final String BLACKLISTED_IC9_PROCEDURE_CODE = "94.45";
+  private static final String BLACKLISTED_IC10_PROCEDURE_CODE = "HZ2ZZZZ";
+  private static final String NON_BLACKLISTED_IC_CODE = "111111";
+
+  private final RifRecordBase claim;
+  private final ExplanationOfBenefit loadedExplanationOfBenefit;
+
+  /** Sets up the test. */
+  @Before
+  public void setup() {
+    samhsaMatcherV2 = new R4SamhsaMatcher();
+  }
+
+  /**
+   * Data collection.
+   *
+   * @return the collection
+   */
+  @Parameterized.Parameters(name = "{0}")
+  public static Collection<Object[]> data() {
+    // Load and transform the various claim types for testing
+    RifRecordBase inpatientClaim = getClaim(InpatientClaim.class);
+    ExplanationOfBenefit inpatientEob =
+        InpatientClaimTransformerV2.transform(
+            new MetricRegistry(), inpatientClaim, Optional.empty());
+
+    RifRecordBase outpatientClaim = getClaim(OutpatientClaim.class);
+    ExplanationOfBenefit outpatientEob =
+        OutpatientClaimTransformerV2.transform(
+            new MetricRegistry(), outpatientClaim, Optional.empty());
+
+    RifRecordBase dmeClaim = getClaim(DMEClaim.class);
+    ExplanationOfBenefit dmeEob =
+        DMEClaimTransformerV2.transform(new MetricRegistry(), dmeClaim, Optional.empty());
+
+    RifRecordBase hhaClaim = getClaim(HHAClaim.class);
+    ExplanationOfBenefit hhaEob =
+        HHAClaimTransformerV2.transform(new MetricRegistry(), hhaClaim, Optional.empty());
+
+    RifRecordBase hospiceClaim = getClaim(HospiceClaim.class);
+    ExplanationOfBenefit hospiceEob =
+        HospiceClaimTransformerV2.transform(new MetricRegistry(), hospiceClaim, Optional.empty());
+
+    RifRecordBase snfClaim = getClaim(SNFClaim.class);
+    ExplanationOfBenefit snfEob =
+        SNFClaimTransformerV2.transform(new MetricRegistry(), snfClaim, Optional.empty());
+
+    RifRecordBase carrierClaim = getClaim(CarrierClaim.class);
+    ExplanationOfBenefit carrierEob =
+        CarrierClaimTransformerV2.transform(new MetricRegistry(), carrierClaim, Optional.empty());
+
+    RifRecordBase pdeClaim = getClaim(PartDEvent.class);
+    ExplanationOfBenefit pdeEob =
+        PartDEventTransformerV2.transform(new MetricRegistry(), pdeClaim, Optional.empty());
+
+    // Load the claim types into the test data that will be run against each test
+    return Arrays.asList(
+        new Object[][] {
+          {inpatientClaim, inpatientEob},
+          {outpatientClaim, outpatientEob},
+          {dmeClaim, dmeEob},
+          {hhaClaim, hhaEob},
+          {hospiceClaim, hospiceEob},
+          {snfClaim, snfEob},
+          {carrierClaim, carrierEob},
+          {pdeClaim, pdeEob}
+        });
+  }
```

Comment:
Ah, this is where all the different types of samhsa codes are checked.  So my question here is just do we need to start with a real claim?  Seems like we only need to worry about the samhsa matcher specific data here?

Reply:
The test is also checking that the output of the ClaimTransformer can be appropriately caught in the matcher, thats why we're ingesting the data from a "real" (test data) claim. The issue we had was related to a mismatch in the code set by the transformer and what we filter on, so this seemed like a good way to enforce that contract.